applications/plugins/SofaPython/Binding_GUIManager.cpp

```diff
@@ -0,0 +1,249 @@
+        std::string msg = "GUIManager.init requires a valid gui name. Registered GUIs are '" + sofa::gui::GUIManager::ListSupportedGUI(',') + "'";
+        SP_MESSAGE_ERROR(msg.c_str());
+        return nullptr;
+    }
+    sofa::gui::GUIManager::Init("sofa", name_ptr);
+    Py_RETURN_NONE;
+}
+
+static constexpr const char* createGUI_DOC =
+R"DOC(
+Create a new GUI.
+
+:return: 1 if the GUI creation failed, 0 otherwise
+:rtype: int
+)DOC";
+static PyObject * GUIManager_createGUI(PyObject * /*self*/, PyObject * /*args*/) {
+    int err = sofa::gui::GUIManager::createGUI();
+    return PyInt_FromLong(err);
+}
+
+static constexpr const char* setDimension_DOC =
+R"DOC(
+Set the GUI's dimension (width and height)
+
+:param width: The GUI's width.
+:param height: The GUI's height.
+:type width: int
+:type height: int
+)DOC";
+static PyObject * GUIManager_setDimension(PyObject * /*self*/, PyObject * args) {
+    unsigned int width;
+    unsigned int height;
+
+    if (!PyArg_ParseTuple(args, "II", &width, &height)) {
+        std::string msg = "GUIManager.setDimension requires two arguments (width and height)";
+        SP_MESSAGE_ERROR(msg.c_str());
+        return nullptr;
+    }
+
+    sofa::gui::GUIManager::SetDimension(width, height);
+
+    Py_RETURN_NONE;
+}
+
+static constexpr const char* setScene_DOC =
+R"DOC(
+Set the GUI's main scene
+
+:param root: The scene's root node
+:param name: (optional) The scene's name
+:type root: sofa::simulation::Node
+:type name: str
+)DOC";
+static PyObject * GUIManager_setScene(PyObject * /*self*/, PyObject * args) {
+    PyObject * node_ptr  = nullptr;
+    const char *filename = nullptr;
+
+    if (!PyArg_ParseTuple(args, "O|s", &node_ptr, &filename)) {
+        std::string msg = "GUIManager.setScene requires a root node";
+        SP_MESSAGE_ERROR(msg.c_str());
+        return nullptr;
+    }
+
+    auto * root = sofa::py::unwrap<sofa::simulation::Node>(node_ptr);
+    sofa::gui::GUIManager::SetScene(root, filename);
+
+    Py_RETURN_NONE;
+}
+
+static constexpr const char* MainLoop_DOC =
+R"DOC(
+Start the GUI's main loop
+
+:param root: The scene's root node
+:param filename: (optional) The scene's filename
+:type root: sofa::simulation::Node
+:type filename: str
+)DOC";
+static PyObject * GUIManager_MainLoop(PyObject * /*self*/, PyObject * args) {
+    PyObject * node_ptr  = nullptr;
+    const char *filename = nullptr;
+
+    if (!PyArg_ParseTuple(args, "O|s", &node_ptr, &filename)) {
+        std::string msg = "GUIManager.MainLoop requires a root node";
+        SP_MESSAGE_ERROR(msg.c_str());
+        return nullptr;
+    }
+
+    auto * root = sofa::py::unwrap<sofa::simulation::Node>(node_ptr);
+    int err = sofa::gui::GUIManager::MainLoop(root, filename);
+
+    return PyInt_FromLong(err);
+}
+
+static PyObject * GUIManager_closeGUI(PyObject * /*self*/, PyObject * /*args*/) {
+    sofa::gui::GUIManager::closeGUI();
+    Py_RETURN_NONE;
+}
+
+
+SP_MODULE_METHODS_BEGIN(GUIManager)
```

Comment:
I'm not found of having the GUIManager in the Sofa module (because this make a dependency from the gui layer into the SofaPython plugin). 

I think it would be better to have it in a separated module that could be named: 
SofaRuntime or SofaApplication
And then have 
SofaApplication.GUIManager or SofaRuntime.GUIManager in it. 

As there is no dependency to the existing binding's class and object I think using pybind11 for such one should be prefered.